### PR TITLE
Fix PodGC test when PodDisruptionConditions disabled

### PR DESCRIPTION
**Pull request by kannon92**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
PodGC says that test should have `PodCondition` turned off but test has it enabled.
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/integration/podgc/podgc_test.go
Comment (mimowo) [inline test/integration/podgc/podgc_test.go:163]: This is actually a bug in the test that could cause flakiness, as with `enablePodDisruptionConditions: true` the phase is transitioned from Pending to Failed. The only reason the test was passing is that we are using `PollImmediate` which makes the check before PodGC changes the phase. Since the PodGC check period is 500ms (https://github.com/kubernetes/kubernetes/blob/3e2840400880d7e698fdb390a9cb8eef08464ce9/test/integration/podgc/podgc_test.go#L293) it is very unlikely the phase would change actually, causing the test flaky, it is just essentially ignored. 

I think in order to make the test useful we should also change the `PollImmediate` to `Poll` so that we pass a single gc check interval. We could also consider lowering the poll interval to 600ms.

We could also consider cherry-picking the fix for the test, but given failure is very unlikely, I feel there is no such need. @alculquicondor wdyt?

File: test/integration/podgc/podgc_test.go
Comment (mimowo) [inline test/integration/podgc/podgc_test.go:162]:
```suggestion
		"pod has phase unchanged when PodDisruptionConditions disabled": {
```

**Review by mimowo — COMMENTED** [review]
Thanks for raising this. This is actually a bug in the test.

**Comment by mimowo** [comment]
Almost LGTM, but I would prefer to make sure the test fails (or at least is flaky) when the phase is wrong, as pointed out [here](https://github.com/kubernetes/kubernetes/pull/118805#discussion_r1238261772). I would suggest to simply change `PollImmediate` to `Poll` [here](https://github.com/kubernetes/kubernetes/blob/3e2840400880d7e698fdb390a9cb8eef08464ce9/test/integration/podgc/podgc_test.go#L247) to give PodGC time to process the pods based on its [500ms gc interval](https://github.com/kubernetes/kubernetes/blob/3e2840400880d7e698fdb390a9cb8eef08464ce9/test/integration/podgc/podgc_test.go#L293). 

Once this is done, we can consider tuning the intervals to make the test run faster, but this could be a follow up.

**Comment by mimowo** [comment]
/lgtm
I also suggest to retitle to "Fix PodGC test when PodDisruptionConditions disabled"

**Comment by kannon92** [comment]
/retitle Fix PodGC test when PodDisruptionConditions disabled

File: test/integration/podgc/podgc_test.go
Comment (alculquicondor) [reply test/integration/podgc/podgc_test.go:163]: Yeah, cherry-picking is worth.

**Review by alculquicondor — COMMENTED** [review]
/approve

**Comment by kannon92** [comment]
https://github.com/kubernetes/kubernetes/issues/108891

/retest

**Comment by kannon92** [comment]
@alculquicondor can you give this an /approved when you get a chance?  I think your original comment didn't go through.

**Comment by alculquicondor** [comment]
I don't have approval rights for this, I forgot :)

**Review by soltysh — APPROVED** [review]
/approve

File: test/integration/podgc/podgc_test.go
Comment (alculquicondor) [reply test/integration/podgc/podgc_test.go:163]: yes please

**Pull request merged**
